src/handshake.rs

```diff
@@ -141,6 +141,10 @@ pub enum Error {
     UnexpectedHeader(String),
     /// The Sec-WebSocket-Accept header value did not match.
     InvalidSecWebSocketAccept,
+    /// The Host header domain is not allowed,
+    InvalidHost(String),
+    /// The Origin header domain is not allowed,
+    InvalidOrigin(String),
```

Comment:
Do we really need to echo back the host they used? Can't we just have `InvalidOrigin` (or `OriginaDenied`).

Reply:
It's useful if you want to log the error I reckon.

Reply:
same here I like `DeniedOrigin` better, same reason as for `InvalidHost` ^^